sktime/tests/test_all_estimators.py

```diff
@@ -776,8 +775,47 @@ def test_set_params(self, estimator_instance):
         """Check that set_params works correctly."""
         estimator = estimator_instance
         params = estimator.get_params()
-        assert estimator.set_params(**params) is estimator
-        _check_set_params(estimator.__class__.__name__, estimator)
+
+        msg = f"set_params of {type(estimator).__name__} does not return self"
+        assert estimator.set_params(**params) is estimator, msg
+
+        is_equal, equals_msg = deep_equals(
+            estimator.get_params(), params, return_msg=True
+        )
+        msg = (
+            f"get_params result of {type(estimator).__name__} (x) does not match "
+            f"what was passed to set_params (y). Reason for discrepancy: {equals_msg}"
+        )
+        assert is_equal, msg
+
+    def test_set_params_sklearn(self, estimator_class):
```

Comment:
this simply mimicks `sklearn` `check_set_params`, the only change is replacing `np.inf` dummy values with alternative test parameter sets from `get_test_params`. `sklearn` can't do this since it does not store test parameter sets, and assumes `np.inf` is always fine.